connector/src/test/scala/com/basho/riak/spark/rdd/SparkDataframesTest.scala

```diff
@@ -0,0 +1,145 @@
+
+  var sqlContextHolder: SQLContext = null
+  var df: DataFrame = null
+
+  @Before
+  def initializeDF(): Unit = {
+    val sqlContext = new org.apache.spark.sql.SQLContext(sc)
+    import sqlContext.implicits._
+    sqlContextHolder = sqlContext
+    df = sc.riakBucket[TestData](DEFAULT_NAMESPACE.getBucketNameAsString)
+      .queryAll.toDF
+    df.registerTempTable("test")
+  }
+
+  @Test
+  def schemaTest(): Unit = {
+    df.printSchema()
+    val schema = df.schema.map(_.name).toList
+    val fields = universe.typeOf[TestData].members.withFilter(!_.isMethod).map(_.name.toString.trim).toList
+    assertEquals(schema.sorted, fields.sorted)
+  }
+
+  @Test
+  def sqlQueryTest(): Unit = {
+    val sqlResult = sqlContextHolder.sql("select * from test where category >= 'CategoryC'").toJSON.collect
+    val expected = """[
+      {"id":"u4","name":"Chris","age":10,"category":"CategoryC"},
+      {"id":"u6","name":"George","age":50,"category":"CategoryC"}
+      ]"""
+    assertEqualsUsingJSONIgnoreOrder(expected, stringify(sqlResult))
+  }
+
+  @Test
+  def udfTest(): Unit = {
+    sqlContextHolder.udf.register("stringLength", (s: String) => s.length)
+    val udf = sqlContextHolder.sql("select name, stringLength(name) strLgth from test order by strLgth").toJSON.collect
+    val expected = """[
+      {"name":"Ben","strLgth":3},
+      {"name":"John","strLgth":4},
+      {"name":"Mary","strLgth":4},
+      {"name":"Clair","strLgth":5},
+      {"name":"Chris","strLgth":5},
+      {"name":"George","strLgth":6}
+      ]"""
+    assertEqualsUsingJSON(expected, stringify(udf))
+  }
+
+  @Test
+  def grouppingTest(): Unit = {
+    val groupped = df.groupBy("category").count.toJSON.collect
+    val expected = """[
+      {"category":"CategoryA","count":1},
+      {"category":"CategoryB","count":2},
+      {"category":"CategoryC","count":2},
+      {"count":1}
+      ]"""
+    assertEqualsUsingJSONIgnoreOrder(expected, stringify(groupped))
+  }
+
+  @Test
+  def sqlVsFilterTest(): Unit = {
+    val sql = sqlContextHolder.sql("select id, name from test where age >= 50").toJSON.collect
+    val filtered = df.where(df("age") >= 50).select("id", "name").toJSON.collect
+    assertEqualsUsingJSONIgnoreOrder(stringify(sql), stringify(filtered))
+  }
+
+  @Test
+  def saveToRiak(): Unit = {
+    val rdd = sc.parallelize(List("""{"field":"value1"}""", """{"field":"value2"}""", """{"field":"value3"}"""))
+    val dataFrame = sqlContextHolder.read.json(rdd)
+    val riakRdd = dataFrame.toJSON.map {
+      line =>
+        val obj = RiakObjectConversionUtil.to(line)
+        obj.setContentType("application/json")
+        obj
+    }
+    riakRdd.saveToRiak(DEFAULT_NAMESPACE_4STORE)
+
+    val fromRiak = sc.riakBucket[String](DEFAULT_NAMESPACE_4STORE.getBucketNameAsString).queryAll
+    assertEquals(3, fromRiak.count)
+  }
+
+  private def assertEqualsUsingJSON(jsonExpected: String, actual: String): Unit = {
```

Comment:
Better to use [AbstractRiakTest.assertEqualsUsingJSON(..)](https://github.com/basho/spark-riak-connector/blob/develop/test-utils/src/main/scala/com/basho/riak/spark/rdd/AbstractRiakTest.scala#L93)

This function allows to use non-strict JSON, therefore we can get rid of kinda "overquoted" JSON.```scala
 val expected = """[
+      {"category":"CategoryA","count":1},
+      {"category":"CategoryB","count":2},
+      {"category":"CategoryC","count":2},
+      {"count":1}
+      ]"""
```

IN this case stringify functions is also not needed